manifests/09-prometheus-rules.yaml

```diff
@@ -5,8 +5,31 @@ metadata:
   namespace: openshift-image-registry
   annotations:
     capability.openshift.io/name: ImageRegistry
-    release.openshift.io/delete: "true"
     include.release.openshift.io/hypershift: "true"
     include.release.openshift.io/ibm-cloud-managed: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
     include.release.openshift.io/single-node-developer: "true"
+spec:
+  groups:
+    - name: pvc-problem-detector.rules
+      rules:
+      - alert: ImageRegistryStorageReadOnly
+        for: 10m
+        expr: sum without(instance, pod, operation) (rate(imageregistry_storage_errors_total{code="READ_ONLY_FILESYSTEM"}[5m])) > 0
+        labels:
+           kubernetes_operator_part_of: image-registry
+           severity: warning
+        annotations:
+           summary: The image registry storage is read-only and no images will be committed to storage.
+           description: The image registry storage is read-only. Read-only storage affects direct pushes to the image registry, and pull-through proxy caching. In the case of pull-through proxy caching, read-only storage is particularly important because without it the image registry won't be actually caching anything. Please verify your backing storage solution and make sure the volume mounted on the image-registry pods is writable to avoid potential outages.
+           message: The image registry storage is read-only and no images will be committed to storage.
+      - alert: ImageRegistryStorageFull
+        for: 10m
+        expr: sum without(instance, pod, operation) (rate(imageregistry_storage_errors_total{code="DEVICE_OUT_OF_SPACE"}[5m])) > 0
```

Comment:
Do we need a warning for a storage threshold? IE: your at 80% of your storage (or some percentage; configurable in a CR somewhere); that triggers warning before there are issues?

Reply:
cross-commenting from slack:
You are right, that would be a better alert. We don't really have any metrics to alert on that though.
The original ask was to alert on read-only storage, the storage full alert is there by request of the PM to alert on any catastrophic error that we could easily detect with current metrics.
I'd prefer to go with what we have in the PR, or to remove the storage full alert entirely. We'd still be covering the customer request by delivering only the read-only alert.
What do you think?

Reply:
PVs getting full should be covered by the `KubePersistentVolumeFillingUp` alerting rules.
 https://github.com/openshift/cluster-storage-operator/blob/ad3c5b5427d86ce2dcfdf83498290bec9482b5e5/manifests/12_prometheusrules.yaml#L74-L118